JM/CF/Scripts/3_Game/CommunityFramework/Timer/CF_Timer.c

```diff
@@ -0,0 +1,55 @@
+class CF_Timer : CF_TimerBase
+{
+	protected Managed m_Object;
+	protected string m_Function;
+	protected ref CF_TimerParam m_Params;
+	
+	void CF_Timer()
+	{
+	}
+
+	static CF_TimerBase Create(Managed obj, string funcName, float duration = 0.025, CF_TimerParam params = null)
+	{
+		#ifdef CF_TRACE_ENABLED
+		CF_Trace trace(null, "Create", "" + obj, "" + funcName, "" + duration, "" + params);
+		#endif
+
+		if (obj == null) return null;
+
+		CF_Timer timer = new CF_Timer();
+		timer.m_Object = obj;
+		timer.m_Function = funcName;
+		timer.m_Params = params;
+		if (!timer.m_Params) timer.m_Params = new CF_TimerParam();
```

Comment:
A bit cleaner imo
```cs
if(!params) params = new CF_TimerParam();
timer.m_Params = params;
```

Why do we assign empty new params? Can't we just invoke the timer function with null? If people did not care about having them in their Create method then they could omit it from their function they subscribe for the OnTick and not care about it. Also saves on performance.

Reply:
The function call contains `CF_TimerBase,float` before the rest of the params continues. That is why.